randovania/games/prime1/gui/preset_settings/prime_enemy_stat_randomizer.py

```diff
@@ -0,0 +1,188 @@
+        self.maximum_label.setEnabled(checked)
+        self.scale_attribute_label.setEnabled(checked)
+        self.range_scale_low.setEnabled(checked)
+        self.range_scale_high.setEnabled(checked)
+        self.health_attribute_label.setEnabled(checked)
+        self.range_health_low.setEnabled(checked)
+        self.range_health_high.setEnabled(checked)
+        self.speed_attribute_label.setEnabled(checked)
+        self.range_speed_low.setEnabled(checked)
+        self.range_speed_high.setEnabled(checked)
+        self.damage_attribute_label.setEnabled(checked)
+        self.range_damage_low.setEnabled(checked)
+        self.range_damage_high.setEnabled(checked)
+        self.knockback_attribute_label.setEnabled(checked)
+        self.range_knockback_low.setEnabled(checked)
+        self.range_knockback_high.setEnabled(checked)
+        self.diff_xyz.setEnabled(checked)
+        self.label.setEnabled(checked)
+        self.label_2.setEnabled(checked)
+
+    def on_preset_changed(self, preset: Preset):
+        config = preset.configuration
+        if config.enemy_attributes:
+            with self._editor as editor:
+                self.range_scale_low.setValue(config.enemy_attributes.enemy_rando_range_scale_low)
+                self.range_scale_high.setValue(config.enemy_attributes.enemy_rando_range_scale_high)
+                self.range_health_low.setValue(config.enemy_attributes.enemy_rando_range_health_low)
+                self.range_health_high.setValue(config.enemy_attributes.enemy_rando_range_health_high)
+                self.range_speed_low.setValue(config.enemy_attributes.enemy_rando_range_speed_low)
+                self.range_speed_high.setValue(config.enemy_attributes.enemy_rando_range_speed_high)
+                self.range_damage_low.setValue(config.enemy_attributes.enemy_rando_range_damage_low)
+                self.range_damage_high.setValue(config.enemy_attributes.enemy_rando_range_damage_high)
+                self.range_knockback_low.setValue(config.enemy_attributes.enemy_rando_range_knockback_low)
+                self.range_knockback_high.setValue(config.enemy_attributes.enemy_rando_range_knockback_high)
+                self.diff_xyz.setChecked(config.enemy_attributes.enemy_rando_diff_xyz)
+
+
+    def _on_spin_changed_range_scale_low(self):
+        with self._editor as editor:
+            config = editor.configuration.enemy_attributes
+            config = dataclasses.replace(config, enemy_rando_range_scale_low=self.range_scale_low.value())
+            editor.set_configuration_field(
+                "enemy_attributes",
+                config,
+            )
+
+    def _on_spin_changed_range_scale_high(self):
+        with self._editor as editor:
+            config = editor.configuration.enemy_attributes
+            config = dataclasses.replace(config, enemy_rando_range_scale_high=self.range_scale_high.value())
+            editor.set_configuration_field(
```

Comment:
You can minimise repeated code by making a helper function that takes a Enemy Attributes object and does the with editor set configuration field stuff